From 337aed75085f548a1f3cc72a508cc75cb2f831e4 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sun, 25 Sep 2022 17:53:05 +0300
Subject: [PATCH] [SW-0] boot_desc: remove die id check from boot_desc_init

Currently, the check of die id is kept inside boot_desc_init
which skips for slave die (die_id=1). However the boot desc
failure flag is set for slave die. Since boot_desc_init is called
only from 2 places for gaudi3, this patch removes check of die
from boot_desc_init function and adds it in caller function.

Change-Id: I9f87770554a28c6acd87c4f03336de221fd1b720
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index a3a9fd7..77357fe 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -474,6 +474,9 @@
 
 	spi_flash_init();
 
+	if (arcboot_rom_global.die_id)
+		return;
+
 	if (!boot_desc_init(HL_FW_BOOT_PTR_SPI))
 		arcboot_rom_status_update(PROGRESS_INIT_SPI_BOOT_DESC_FAIL);
 }
diff --git a/common/boot_desc.c b/common/boot_desc.c
index ee2de50..aed4986 100644
--- a/common/boot_desc.c
+++ b/common/boot_desc.c
@@ -75,9 +75,6 @@
 	int err;
 	static struct hl_fw_img_desc *p_img_desc;
 
-	if (arcboot_rom_global.die_id)
-		return 0;
-
 	if (img >= HL_FW_BOOT_PTR_MAX_IMG)
 		return 0;
 
